optical_flow.rs

```diff
@@ -0,0 +1,127 @@
+use crate::spi::HypedSpi;
+use std::time::{Duration, Instant, SystemTime, UNIX_EPOCH};
+use byteorder::{LittleEndian, ReadBytesExt};
+use std::thread::sleep;
+use serde::{Serialize, Deserialize};
+use serde_json::json;
+
+
+// implement get_motion function first
+// add readByteFromRegister (?) function to HypedSPI trait which takes register address and gives data
+
+const REG_MOTION_BURST: u8 = 0x16;
+const TIMEOUT: Duration = Duration::from_secs(5);
+
+
+pub struct PMW3901<SPI> {
+    spi: SPI,
+
+}
+
+impl<SPI> PMW3901<SPI>
+where
+    SPI: HypedSpi,
+{
```

Comment:
```suggestion
impl<SPI: HypedSpi> PMW3901<SPI>
```
You should be able to do something shorter like this instead